src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -61,12 +69,48 @@ EffectManifestPointer PitchShiftEffect::getManifest() {
     pitch->setNeutralPointOnScale(0.0);
     pitch->setRange(-1.0, 0.0, 1.0);
 
+    EffectManifestParameterPointer range = pManifest->addParameter();
+    range->setId("range");
+    range->setName(QObject::tr("Range"));
+    range->setShortName(QObject::tr("Range"));
+    range->setDescription(QObject::tr(
+            "The range of the Pitch knob (0 - 2 octaves).\n"));
+    range->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
+    range->setDefaultLinkType(EffectManifestParameter::LinkType::Linked);
+    range->setNeutralPointOnScale(1.0);
+    range->setRange(0.0, 1.0, 2.0);
+
+    EffectManifestParameterPointer semitonesMode = pManifest->addParameter();
+    semitonesMode->setId("semitonesMode");
+    semitonesMode->setName(QObject::tr("Semitones"));
+    semitonesMode->setShortName(QObject::tr("Semitones"));
+    semitonesMode->setDescription(QObject::tr(
+            "Change the pitch in semitone steps instead of continuously."));
+    semitonesMode->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    semitonesMode->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    semitonesMode->setRange(0, 1, 1);
+
+    EffectManifestParameterPointer formantPreserving = pManifest->addParameter();
+    formantPreserving->setId("formantPreserving");
+    formantPreserving->setName(QObject::tr("Formant"));
+    formantPreserving->setShortName(QObject::tr("Formant"));
+    formantPreserving->setDescription(QObject::tr(
+            "Preserve the resonant frequencies (formants) of the human vocal tract "
+            "and other instruments.\n"
+            "Hint: compensates \"chipmunk\" or \"growling\" voices"));
+    formantPreserving->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    formantPreserving->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    formantPreserving->setRange(0, 0, 1);
+
     return pManifest;
 }
 
 void PitchShiftEffect::loadEngineEffectParameters(
         const QMap<QString, EngineEffectParameterPointer>& parameters) {
     m_pPitchParameter = parameters.value("pitch");
+    m_pRangeParameter = parameters.value("range");
+    m_pSemitonesModeParameter = parameters.value("semitonesMode");
+    m_pFormantPreservingParameter = parameters.value("formant");
```

Comment:
```suggestion
    m_pFormantPreservingParameter = parameters.value("formantPreserving");
```
This fixes the segfault.

Reply:
Oh, thank you. I'm sorry, I overlooked it.